src/app-common/tree-menu/walk.js

```diff
@@ -2,9 +2,27 @@ const validateData = (data) => !!data && data.length > 0;
 
 const getValidatedData = (data) => (validateData(data) ? data : []);
 
-export const walk = ({ data, ...props }) => {
-  const validatedData = getValidatedData(data);
+const filter = ({ typeFilter, data }) => {
+  const leafNodes = [];
+  const typeKey = typeFilter === "STREAM_LOCATION" ? "sub_location_type" : "location_type";
 
+  const getLeafNodes = ( leafNodes, obj ) => {
+    if ( !obj.is_leaf ) {
+      obj.nodes.forEach(function (child) {
+        getLeafNodes(leafNodes, child);
+      });
+    } else if ( obj[typeKey] === typeFilter ) {
+      leafNodes.push( obj );
+    }
+    return leafNodes;
+  };
+  return getLeafNodes( leafNodes, data[0] );
+};
+
+export const walk = ({ data, typeFilter, ...props }) => {
+  let treeData;
+  typeFilter ? ( treeData = filter({ typeFilter, data }).sort((a, b) => a.label.trim().localeCompare(b.label.trim())) ) : ( treeData = data );
```

Comment:
Is sorting the items when filter is on just to make it easier for them to find something in the list? Meaning that this would be removed if we figure out how to maintain the parent tree structure during filtering?

Reply:
Yep, exactly.